lib/incoming_mail.rb

```diff
@@ -0,0 +1,110 @@
+module IncomingMail
+  # mail_types
+  UNKNOWN = 0
+  REQUEST = 1
+
+  class Handler
+    # parses a Mandrill callback object and Does The Right Thing
+
+    def initialize(params = {})
+      params.each do |key, value|
+        instance_variable_set("@#{key}", value)
+      end
+    end
+
+    def handle(payload)
+      if payload.is_a?(Array) and payload[0]['event'] and payload[0]['event'] == 'inbound'
+        create_response(payload[0])
+      elsif payload.is_a?(Mandrill::WebHook::EventDecorator)
+        create_response(payload)
+      else
+        fail "Invalid Mandrill event payload. Must be event==inbound"
+      end
+    end
+
+    private
+
+    def create_response(payload)
+      resp = Response.new(type: identify_mail_type(payload))
+      case resp.type
+      when REQUEST
+        resp.comment = create_comment(payload['msg'])
+        resp.action = resp.comment ? Response::COMMENT : Response::ERROR
+      else
+        resp.action = Response::DROPPED
+      end
+      resp
+    end
+
+    def identify_mail_type(payload)
+      subject = payload['msg']['subject']
+      references = payload['msg']['headers']['References']
+      if subject.match(/Request (#|FY)\d+/)
+        return IncomingMail::REQUEST
+      elsif references and references.match(/<proposal-\d+/)
+        return IncomingMail::REQUEST
+      else
+        return IncomingMail::UNKNOWN
+      end
+    end
+
+    def create_comment(msg)
+      proposal     = find_proposal(find_public_id(msg)) or return
```

Comment:
is aligning the `=`s here a style we are using in this project? (I usually do not do this but am unsure what this alignment is called and therefore am having trouble googling for rubocop related rules)